// react-native-table-component-pro/components/cell.js
import React, { Component } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

class Cell extends Component {
  static propTypes = {
    style: View.propTypes.style,
    textStyle: Text.propTypes.style,
    borderStyle: View.propTypes.style,
  }

  render() {
    const {data, width, height, flex, style, textStyle} = this.props;
    let borderWidth,borderColor;
    if (this.props.borderStyle && this.props.borderStyle.borderWidth) {
      borderWidth = this.props.borderStyle.borderWidth;
    } else {
      borderWidth = 0.5;
    }
    if (this.props.borderStyle && this.props.borderStyle.borderColor) {
      borderColor = this.props.borderStyle.borderColor;
    } else {
      borderColor = '#000';
    }
    return (
      <View style={[
        {
          borderTopWidth: borderWidth,
          borderRightWidth: borderWidth,
          borderColor: borderColor,
        },
        styles.cell,
        width && {width: width},
        height && {height: height},
        flex && {flex: flex},
        !width && !flex && !height && {flex: 1},
        style
      ]}>
        {typeof data === "string" && <Text style={[textStyle, styles.text]}>{data}</Text>}
        {typeof data === "object" &&
          <TouchableOpacity
            onPress={()=>{
              data.onPress()
            }}
          >
            {data.hasOwnProperty("text") && <Text style={[textStyle, styles.text]}>{data.text}</Text>}
            {data.hasOwnProperty("icon") && <Text style={[textStyle, styles.text]}><Icon name={data.icon} color={data.iconColor} size={data.iconSize}/></Text>}
          </TouchableOpacity>
        }
      </View>
    )
  }
}

const styles = StyleSheet.create({
  cell: {
    justifyContent: 'center',
  },
  text: {
    backgroundColor: 'transparent',
  },
})

export default Cell;
